data/api_v1_middleware.go

```diff
@@ -132,26 +149,40 @@ func (a *API) middlewareV1(fn HandlerLoggerFunc, params ...string) http.HandlerF
 				// userID is a commonly used parameter
 				// See if we can view the data
 				userID := res.VARS["userID"]
-
-				if userID == "" {
-					res.WriteError(&errorInvalidParameters)
-				} else if !a.isAuthorized(r, []string{userID}) {
-					res.WriteError(&errorNoViewPermission)
+				userIDs = []string{userID}
+
+				if !regexpUserID.MatchString(userID) {
```

Comment:
Why do we add this pattern matching ? 
If the userId doesn't have the expected pattern, we won't find any documents, and we won't even authorize it through coastguard. 

On the other hand we have to do a regexp match on all valid userIds which has a compute cost. And if one day we change userId format in shoreline we'll have to change the regexp in tide-whisperer also...

Reply:
It's to verify the input, since we can't trust theses values. For security reason.